### PR TITLE
Entities, Blocks, Chunks, ordering-before-naming and more

### PR DESCRIPTION
**Pull request by ENDERZOMBI102**

### CODE REVIEW

**Review by PseudoDistant — APPROVED** [review]
Looks good.

File: mappings/net/minecraft/achievement/class_3348.mapping [hunk 1]
```diff
@@ -0,0 +1 @@
+CLASS net/minecraft/class_3348 net/minecraft/achievement/class_3348
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/achievement/class_3348.mapping:1]: why

File: mappings/net/minecraft/achievement/class_3363.mapping [hunk 2]
```diff
@@ -0,0 +1 @@
+CLASS net/minecraft/class_3363 net/minecraft/achievement/class_3363
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/achievement/class_3363.mapping:1]: why

File: mappings/net/minecraft/achievement/class_3366.mapping [hunk 3]
```diff
@@ -0,0 +1 @@
+CLASS net/minecraft/class_3366 net/minecraft/achievement/class_3366
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/achievement/class_3366.mapping:1]: why

File: mappings/net/minecraft/achievement/class_3370.mapping [hunk 4]
```diff
@@ -0,0 +1 @@
+CLASS net/minecraft/class_3370 net/minecraft/achievement/class_3370
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/achievement/class_3370.mapping:1]: why

File: mappings/net/minecraft/achievement/class_3376.mapping [hunk 5]
```diff
@@ -0,0 +1 @@
+CLASS net/minecraft/class_3376 net/minecraft/achievement/class_3376
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/achievement/class_3376.mapping:1]: why

**Comment by ENDERZOMBI102** [comment]
whats the problem with the build?

**Review by BoogieMonster1O1 — COMMENTED** [review]
there's a few unmapped classes with nothing in them
i'll do a full review later

**Review by PseudoDistant — COMMENTED** [review]
Seems it still needs work...
Sorry, it's been a good bit since I've worked with mappings...

File: mappings/net/minecraft/world/dimension/Dimension.mapping [hunk 6]
```diff
@@ -3,13 +3,22 @@ CLASS net/minecraft/class_1208 net/minecraft/world/dimension/Dimension
 	FIELD field_4785 world Lnet/minecraft/class_1150;
 	FIELD field_4786 generatorType Lnet/minecraft/class_1160;
 	FIELD field_4787 biomeSource Lnet/minecraft/class_1175;
+	FIELD field_4787 singletonBiomeSource Lnet/minecraft/class_1178;
```
Comment (arthurbambou) [inline mappings/net/minecraft/world/dimension/Dimension.mapping:6]: this issue again... the key got duplicated and the mappings build doesn't try to only use the good one...

**Review by arthurbambou — CHANGES_REQUESTED** [review]
one of `field_4787` must be removed for now until we are able to handle this.

**Comment by ENDERZOMBI102** [comment]
> one of `field_4787` must be removed for now until we are able to handle this.

there was a method with the same sighnature and intermediary name too

**Comment by arthurbambou** [comment]
methods aren't affected by this issue, only fields

**Comment by ENDERZOMBI102** [comment]
how could it differentiate tho?
wouldn't the same intermediary name conflict? both impls didn't have parameters

**Review by arthurbambou — CHANGES_REQUESTED** [review]
Another conflict to fix:
image

File: mappings/net/minecraft/world/chunk/ServerChunkProvider.mapping [hunk 7]
```diff
@@ -12,13 +13,32 @@ CLASS net/minecraft/class_794 net/minecraft/world/chunk/ServerChunkProvider
...
 	METHOD method_2127 scheduleUnload (II)V
 		ARG 1 y
 		ARG 2 z
+	METHOD method_3864 (II)Z
+		ARG 1 x
+		ARG 2 z
 	METHOD method_3871 getOrGenerateChunk (II)Lnet/minecraft/class_1196;
 		ARG 1 x
 		ARG 2 z
+	METHOD method_4727 save ()V
```
Comment (arthurbambou) [inline mappings/net/minecraft/world/chunk/ServerChunkProvider.mapping:41]: rename this one to `flushChunks` or rename the other implementation in class_1193, class_1214, class_1219, class_476, class_1218 and class_1213 to the same name.

**Review by arthurbambou — CHANGES_REQUESTED** [review]
A way to fix the conflict

**Comment by arthurbambou** [comment]
> > there's a few unmapped classes with nothing in them
> > i'll do a full review later
> 
> i saw various classes in the repo that already were mapped only in the package, but not in the name, so i thought it was ok oh well, i'll map them tomorrow then, not a problem : P

I did it, but no one seemed to care about it :thinking: .

File: mappings/net/minecraft/world/chunk/ServerChunkProvider.mapping (hunk 7 above)
Comment (ENDERZOMBI102) [reply mappings/net/minecraft/world/chunk/ServerChunkProvider.mapping:41]: done

**Review by arthurbambou — APPROVED** [review]

File: mappings/net/minecraft/entity/LlamaEntity.mapping [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+	FIELD field_15511 STRENGHT Lnet/minecraft/class_2921;
+	FIELD field_15513 VARIANT Lnet/minecraft/class_2921;
+	METHOD method_14024 getStrenght ()I
+	METHOD method_14025 getVariant ()I
+	METHOD method_14033 attack (Lnet/minecraft/class_1699;)V
+		ARG 1 entity
+	METHOD method_14035 setVariant (I)V
+		ARG 1 variant
+	METHOD method_14036 setStrenght (I)V
+		ARG 1 strenght
```
Comment (BleachDrinker420) [inline mappings/net/minecraft/entity/LlamaEntity.mapping:11]: Typos
```suggestion
	FIELD field_15511 STRENGTH Lnet/minecraft/class_2921;
	FIELD field_15513 VARIANT Lnet/minecraft/class_2921;
	METHOD method_14024 getStrength ()I
	METHOD method_14025 getVariant ()I
	METHOD method_14033 attack (Lnet/minecraft/class_1699;)V
		ARG 1 entity
	METHOD method_14035 setVariant (I)V
		ARG 1 variant
	METHOD method_14036 setStrength (I)V
		ARG 1 strength
```

File: mappings/net/minecraft/item/ItemStack.mapping [hunk 9]
```diff
@@ -1,7 +1,10 @@
 CLASS net/minecraft/class_1071 net/minecraft/item/ItemStack
+	FIELD field_12184 lastDestroyedBlock Lnet/minecraft/class_197;
 	FIELD field_12185 lastDestroyResult Z
+	FIELD field_12186 lastPlacedOn Lnet/minecraft/class_197;
 	FIELD field_12187 lastPlaceOnResult Z
 	FIELD field_15108 EMPTY Lnet/minecraft/class_1071;
+	FIELD field_15109 invalid Z
```
Comment (BleachDrinker420) [inline mappings/net/minecraft/item/ItemStack.mapping:7]:
```suggestion
	FIELD field_15109 empty Z
```

File: mappings/net/minecraft/item/ItemStack.mapping [hunk 10]
```diff
@@ -56,7 +66,46 @@ CLASS net/minecraft/class_1071 net/minecraft/item/ItemStack
...
+	METHOD method_11395 equals (Lnet/minecraft/class_1071;)Z
+		ARG 1 other
+	METHOD method_11396 (Lnet/minecraft/class_2232;)Z
+		ARG 1 state
+	METHOD method_11397 equals (Lnet/minecraft/class_1071;Lnet/minecraft/class_1071;)Z
+		ARG 0 stack0
+		ARG 1 stack1
+	METHOD method_13652 getCount ()I
+	METHOD method_13653 updateStackStatus ()V
+	METHOD method_13654 isInvalid ()Z
```
Comment (BleachDrinker420) [inline mappings/net/minecraft/item/ItemStack.mapping:102]:
```suggestion
	METHOD method_13654 isEmpty ()Z
```

File: mappings/net/minecraft/nbt/AbstractNumberTag.mapping [hunk 11]
```diff
@@ -0,0 +1,7 @@
+CLASS net/minecraft/class_1947 net/minecraft/nbt/AbstractNumberTag
+	METHOD method_7371 getLong ()J
+	METHOD method_7372 getInt ()I
+	METHOD method_7373 getShort ()S
+	METHOD method_7374 getByte ()B
+	METHOD method_7375 getDouble ()D
+	METHOD method_7376 getFLoat ()F
```
Comment (BleachDrinker420) [inline mappings/net/minecraft/nbt/AbstractNumberTag.mapping:7]: Typo
```suggestion
	METHOD method_7376 getFloat ()F
```

File: mappings/net/minecraft/nbt/Tag.mapping [hunk 12]
```diff
@@ -2,18 +2,20 @@ CLASS net/minecraft/class_605 net/minecraft/nbt/Tag
...
 	METHOD method_1649 read (Ljava/io/DataInput;ILnet/minecraft/class_1944;)V
 		ARG 1 input
 		ARG 2 depth
 		ARG 3 tracker
 	METHOD method_1650 write (Ljava/io/DataOutput;)V
 		ARG 1 output
-	METHOD method_7370 asString ()Ljava/lang/String;
 	METHOD method_1651 copy ()Lnet/minecraft/class_605;
 	METHOD method_4386 isEmpty ()Z
+	METHOD method_7369 createByTypeId (B)Lnet/minecraft/class_605;
```
Comment (BleachDrinker420) [inline mappings/net/minecraft/nbt/Tag.mapping:16]: Awkward name
```suggestion
	METHOD method_7369 createFromType (B)Lnet/minecraft/class_605;
```

File: mappings/net/minecraft/world/gen/feature/EndExitPortalFeature.mapping [hunk 13]
```diff
@@ -0,0 +1,2 @@
+CLASS net/minecraft/class_2755 net/minecraft/world/gen/feature/EndExitPortalFeature
+	FIELD field_12981 doNeightbourBlockUpdates Z
```
Comment (BleachDrinker420) [inline mappings/net/minecraft/world/gen/feature/EndExitPortalFeature.mapping:2]:
```suggestion
	FIELD field_12981 open Z
```

**Review by BleachDrinker420 — CHANGES_REQUESTED** [review]
My review

**Review by PseudoDistant — APPROVED** [review]
thonk

File: mappings/net/minecraft/client/Minecraft.mapping [hunk 14]
```diff
@@ -29,6 +32,7 @@ CLASS net/minecraft/client/Minecraft net/minecraft/client/Minecraft
 	FIELD field_3772 debugTime J
 	FIELD field_3773 currentServerEntry Lnet/minecraft/class_485;
 	FIELD field_3774 instance Lnet/minecraft/class_1600;
+	FIELD field_3774 _instance Lnet/minecraft/client/Minecraft;
```
Comment (arthurbambou) [inline mappings/net/minecraft/client/Minecraft.mapping:35]:
```suggestion
```

File: mappings/net/minecraft/client/Minecraft.mapping [hunk 15]
```diff
@@ -47,7 +51,9 @@ CLASS net/minecraft/client/Minecraft net/minecraft/client/Minecraft
 	FIELD field_3794 joinPlayerCounter I
 	FIELD field_3795 isDemo Z
 	FIELD field_3796 clientConnection Lnet/minecraft/class_1957;
+	FIELD field_3796 conection Lnet/minecraft/class_630;
```
Comment (arthurbambou) [inline mappings/net/minecraft/client/Minecraft.mapping:54]:
```suggestion
```

File: mappings/net/minecraft/client/gui/screen/Screen.mapping [hunk 16]
```diff
@@ -5,6 +5,7 @@ CLASS net/minecraft/class_388 net/minecraft/client/gui/screen/Screen
 	FIELD field_10463 clickedLink Ljava/net/URI;
 	FIELD field_1228 prevClickedButton Lnet/minecraft/class_356;
 	FIELD field_1229 client Lnet/minecraft/class_1600;
+	FIELD field_1229 _client Lnet/minecraft/client/Minecraft;
```
Comment (arthurbambou) [inline mappings/net/minecraft/client/gui/screen/Screen.mapping:8]:
```suggestion
```

File: mappings/net/minecraft/client/options/GameOptions.mapping [hunk 17]
```diff
@@ -16,139 +32,178 @@ CLASS net/minecraft/class_347 net/minecraft/client/options/GameOptions
...
+	FIELD field_942 keyPlayerList Lnet/minecraft/class_327;
+	FIELD field_943 keyPickItem Lnet/minecraft/class_327;
+	FIELD field_944 keyCommand Lnet/minecraft/class_327;
 	FIELD field_945 keysAll [Lnet/minecraft/class_327;
 	FIELD field_946 client Lnet/minecraft/class_1600;
-	FIELD field_7667 difficulty Lnet/minecraft/class_2151;
-	FIELD field_973 invertYMouse Z
-	FIELD field_10327 playerModelParts Ljava/util/Set;
-	FIELD field_7656 soundVolumeLevels Ljava/util/Map;
+	FIELD field_946 _client Lnet/minecraft/client/Minecraft;
```
Comment (arthurbambou) [inline mappings/net/minecraft/client/options/GameOptions.mapping:74]:
```suggestion
```

**Review by arthurbambou — CHANGES_REQUESTED** [review]
Remove those until we've found a way to make tiny remapper more clever :sob:

File: mappings/net/minecraft/client/Minecraft.mapping (hunk 14 above)
Comment (arthurbambou) [inline mappings/net/minecraft/client/Minecraft.mapping:35]:
```suggestion
	FIELD field_3774 instance Lnet/minecraft/client/Minecraft;
```

File: mappings/net/minecraft/client/Minecraft.mapping (hunk 15 above)
Comment (arthurbambou) [inline mappings/net/minecraft/client/Minecraft.mapping:54]:
```suggestion
	FIELD field_3796 conection Lnet/minecraft/class_630;
```

**Review by arthurbambou — CHANGES_REQUESTED** [review]

**Review by arthurbambou — CHANGES_REQUESTED** [review]
please merge upstream into your branch

File: mappings/net/minecraft/nbt/Tag.mapping (hunk 12 above)
Comment (ENDERZOMBI102) [reply mappings/net/minecraft/nbt/Tag.mapping]: done

**Review by arthurbambou — CHANGES_REQUESTED** [review]
image

**Comment by ENDERZOMBI102** [comment]
> image

what does that even mean

File: mappings/net/minecraft/item/ItemStack.mapping [hunk 18]
```diff
@@ -56,7 +61,46 @@ CLASS net/minecraft/class_1071 net/minecraft/item/ItemStack
...
+	METHOD method_11395 equals (Lnet/minecraft/class_1071;)Z
+		ARG 1 other
+	METHOD method_11396 (Lnet/minecraft/class_2232;)Z
+		ARG 1 state
+	METHOD method_11397 equals (Lnet/minecraft/class_1071;Lnet/minecraft/class_1071;)Z
+		ARG 0 stack0
+		ARG 1 stack1
+	METHOD method_13652 getCount ()I
+	METHOD method_13653 updateStackStatus ()V
+	METHOD method_13654 isEmpty ()Z
```
Comment (arthurbambou) [inline mappings/net/minecraft/item/ItemStack.mapping:97]: rename this one to anything else

**Review by arthurbambou — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/item/ItemStack.mapping (hunk 18 above)
Comment (arthurbambou) [inline mappings/net/minecraft/item/ItemStack.mapping:97]:
```suggestion
	METHOD method_13654 isEmpty_ ()Z
```

File: mappings/net/minecraft/entity/player/PlayerEntity.mapping [hunk 19]
```diff
@@ -24,27 +19,24 @@ CLASS net/minecraft/class_988 net/minecraft/entity/player/PlayerEntity
 	FIELD field_3990 prevCapeZ D
 	FIELD field_3991 inBed Z
 	FIELD field_3992 pos Lnet/minecraft/class_2552;
+	FIELD field_3998 abilities Lnet/minecraft/class_986;
+	FIELD field_3999 inventory Lnet/minecraft/class_987;
+	FIELD field_4000 playerScreenHandler Lnet/minecraft/class_1002;
 	FIELD field_4001 spawnPos Lnet/minecraft/class_2552;
-	FIELD field_5398 spawnForced Z
-	FIELD field_12123 enchantmentTableSeed I
+	FIELD field_4001 spawnLocation Lnet/minecraft/class_851;
```
Comment (arthurbambou) [inline mappings/net/minecraft/entity/player/PlayerEntity.mapping:26]:
```suggestion
```

File: mappings/net/minecraft/entity/decoration/PaintingEntity.mapping [hunk 20]
```diff
@@ -1,5 +1,22 @@
 CLASS net/minecraft/class_874 net/minecraft/entity/decoration/PaintingEntity
+	FIELD field_3376 rotation I
+	FIELD field_3377 tileX I
+	FIELD field_3378 tileY I
+	FIELD field_3379 tileZ I
 	FIELD field_3380 type Lnet/minecraft/class_874$class_876;
+	FIELD field_3380 motive Lnet/minecraft/class_876;
```
Comment (arthurbambou) [inline mappings/net/minecraft/entity/decoration/PaintingEntity.mapping:7]:
```suggestion
```

File: mappings/net/minecraft/class_469.mapping [hunk 21]
```diff
@@ -1,8 +1,10 @@
 CLASS net/minecraft/class_469
+	FIELD field_1617 stateManager Lnet/minecraft/class_106;
 	FIELD field_1620 random Ljava/util/Random;
 	FIELD field_1621 disconnected Z
 	FIELD field_1622 connection Lnet/minecraft/class_630;
 	FIELD field_1623 client Lnet/minecraft/class_1600;
+	FIELD field_1623 _client Lnet/minecraft/client/Minecraft;
```
Comment (arthurbambou) [inline mappings/net/minecraft/class_469.mapping:7]:
```suggestion
```

File: mappings/net/minecraft/entity/player/ServerPlayerEntity.mapping [hunk 22]
```diff
@@ -1,52 +1,64 @@
...
+CLASS net/minecraft/class_798 net/minecraft/entity/player/ServerPlayerEntity
+	FIELD field_11763 spectatingEntity Lnet/minecraft/class_864;
+	FIELD field_16399 file Lnet/minecraft/class_3347;
 	FIELD field_2823 networkHandler Lnet/minecraft/class_2107;
-	FIELD field_8893 LOGGER Lorg/apache/logging/log4j/Logger;
-	FIELD field_6730 lang Ljava/lang/String;
-	FIELD field_2840 removedEntities Ljava/util/List;
-	FIELD field_8894 statHandler Lnet/minecraft/class_2141;
-	FIELD field_6731 completeHealth F
+	FIELD field_2823 packetListener Lnet/minecraft/class_803;
```
Comment (arthurbambou) [inline mappings/net/minecraft/entity/player/ServerPlayerEntity.mapping:5]:
```suggestion
```

File: mappings/net/minecraft/server/PlayerManager.mapping [hunk 23]
```diff
@@ -1,15 +1,19 @@
...
 	FIELD field_2711 DATE_FORMATTER Ljava/text/SimpleDateFormat;
 	FIELD field_2712 server Lnet/minecraft/server/MinecraftServer;
 	FIELD field_2713 bannedPlayers Lnet/minecraft/class_733;
 	FIELD field_2714 bannedIps Lnet/minecraft/class_733;
+	FIELD field_2715 ADMINS Ljava/util/Set;
+	FIELD field_2716 _whitelist Ljava/util/Set;
 	FIELD field_2717 saveHandler Lnet/minecraft/class_105;
 	FIELD field_2718 whitelistEnabled Z
 	FIELD field_2719 gameMode Lnet/minecraft/class_1156$class_1157;
+	FIELD field_2719 gamemode Lnet/minecraft/class_1157;
```
Comment (arthurbambou) [inline mappings/net/minecraft/server/PlayerManager.mapping:16]:
```suggestion
```

**Review by arthurbambou — APPROVED** [review]
Don't add anything more to this pr, it's already really big

**Comment by arthurbambou** [comment]
I fixed conflicts so it's building now

**Pull request merged**